### PR TITLE
♻️Rabbitmq rpc: response shall not be jsonized

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
- [x] do **not** jsonize the RPC response.
- [x] have a RPCServerError

The concept is to have a common/shared interface module that defines the types of whatever is returned through RPC

## Related issue/s
pre-requisite of https://github.com/ITISFoundation/osparc-simcore/pull/4703

### CODE REVIEW

File: packages/service-library/src/servicelib/rabbitmq/_rpc_router.py [hunk 1]
```diff
@@ -32,14 +32,18 @@ async def wrapper(*args, **kwargs):
...
-                            result,
-                            custom_encoder=_RPC_CUSTOM_ENCODER,
-                        )
-                    )
+                ):
+                    try:
+                        result = await func(*args, **kwargs)
+                        return result
+                    except asyncio.CancelledError:
+                        _logger.debug("call was cancelled")
```
Comment (pcrespov) [inline packages/service-library/src/servicelib/rabbitmq/_rpc_router.py:40]: MINOR: some extra info?

**Review by pcrespov — APPROVED** [review]

**Comment by pcrespov** [comment]
:tada:  will start using this in my new PR! thx!

File: packages/service-library/src/servicelib/rabbitmq/_rpc_router.py (hunk 1 above)
Comment (sanderegg) [reply packages/service-library/src/servicelib/rabbitmq/_rpc_router.py:40]: for cancellation?

**Review by GitHK — APPROVED** [review]
👍

**Pull request merged**